src/screens/WalletConnect/Requests/WalletConnectSwitchNetwork.tsx

```diff
@@ -0,0 +1,287 @@
+          <Text variant="regular" color={item.isDeployed ? colors.positive : colors.negative}>
+            {item.isDeployed ? tRoot('label.deployed') : tRoot('label.notDeployed')}
+          </Text>
+        </>
+      )}
+      {type === 'selectedChain' && isActiveEtherspotAccount && (
+        <Icon
+          style={[styles.upDnIcon, { backgroundColor: colors.basic050 }]}
+          name={!showList ? 'chevron-down' : 'chevron-up'}
+        />
+      )}
+    </TouchableOpacity>
+  );
+
+  const ListModal = () => (
+    <RNModal animationType="none" visible={showList} transparent>
+      <TouchableOpacity
+        activeOpacity={1}
+        style={{ flex: 1 }}
+        onPress={() => {
+          setShowList(false);
+        }}
+      />
+      <View
+        style={[styles.mainContent, { backgroundColor: colors.basic60, top: dropDownFromTop }, styles.bottomRadius]}
+      >
+        <FlatList
+          contentContainerStyle={styles.listContainer}
+          data={chains}
+          renderItem={({ item }) => renderItem({ item, type: 'list' })}
+          keyExtractor={(item) => item.value}
+          indicatorStyle="white"
+        />
+      </View>
+    </RNModal>
+  );
+
+  return (
+    <>
+      {isActiveEtherspotAccount && !selectedNetwork.isDeployed && (
+        <Text style={[styles.warning, { color: colors.helpIcon }]}>
+          {t('paragraph.undeploy_warning_1')}
+          <Text style={{ color: colors.tertiaryText }}>{t('paragraph.deploy_manager')}</Text>
+          {t('paragraph.undeploy_warning_2')}
+        </Text>
+      )}
+
+      {isActiveArchanovaAccount && (
+        <Text style={[styles.warning, { color: colors.helpIcon }]}>{t('paragraph.archanova_warning')}</Text>
+      )}
+
+      <FlatList
+        key="walletconnect_accounts"
+        data={accounts}
+        horizontal
+        bounces={false}
+        ItemSeparatorComponent={() => <Spacing w={8} />}
+        renderItem={({ item }) => (
+          <TouchableOpacity
+            style={[styles.walletBtn, activeAccount?.id === item.id && { backgroundColor: colors.modalHandleBar }]}
+            onPress={() => dispatch(switchAccountAction(item.id))}
+            key={item.id}
+          >
+            <Text style={styles.walletTxt}>{item.label}</Text>
+          </TouchableOpacity>
+        )}
+        keyExtractor={(item) => item.id}
+      />
+
+      <View
+        ref={btnRef}
+        onLayout={(event) => {
+          const { height } = event.nativeEvent.layout;
+          if (contentHeight !== height) setContentHeight(height);
+        }}
+        style={[styles.container, { backgroundColor: colors.basic60 }, !showList && styles.bottomRadius]}
+      >
+        {renderItem({ item: selectedNetwork, type: 'selectedChain' })}
+
+        {showList && <View style={[styles.line, { backgroundColor: colors.basic050 }]} />}
+      </View>
+
+      <ListModal />
+    </>
+  );
+};
+
+export default WalletConnectSwitchNetwork;
+
+const useChains = (): any[] => {
```

Comment:
Could we please move this function to the top of the render method for consistency

Reply:
Updated.